cirqtools/qudit/common_gates.py

```diff
@@ -136,3 +138,51 @@ def _circuit_diagram_info_(self, args):
             return '[{}<->{}]'.format(self.flip_a, self.flip_b)
         return cirq.CircuitDiagramInfo((
             '[{}<->{}]'.format(self.flip_a, self.flip_b),))
+
+class Chrestenson(cirq.SingleQubitGate):
+    def __init__(self,dimension):
+        self.dimension = dimension
+
+    def _qid_shape_(self):
+        return (self.dimension,)
+
+    def num_qubits(self):
+        return 1
```

Comment:
This isn't needed when `_qid_shape_` is given.

Reply:
change made!